mtp_transaction_uploader/settings.py

```diff
@@ -54,7 +54,14 @@
 NOMS_AGENCY_ACCOUNT_NUMBER = os.environ.get('NOMS_AGENCY_ACCOUNT_NUMBER', '67175315')
 NOMS_AGENCY_SORT_CODE = os.environ.get('NOMS_AGENCY_SORT_CODE', '123456')
 
-WORLDPAY_SETTLEMENT_REFERENCE = os.environ.get('WORLDPAY_SETTLEMENT_REFERENCE', '.*GGGGGGGG.*([0-9]{4}).*')
+# WorldPay settlements can be in the form:
+# - "PREFIX0101" with the last 4 digits being a day/month
+# - "PREFIX01" with the last 2 digits being a day
+# - "PREFIX" with no date reference
+# where the dates refer to the day the settlement is for
+# (i.e. when the money was collected from the sender / transaction was captured)
+# this regular expression must have a capturing group called 'date' which accepts 0, 2 or 4 digits
+WORLDPAY_SETTLEMENT_REFERENCE = os.environ.get('WORLDPAY_SETTLEMENT_REFERENCE', 'TT- GGGGGGGG -(?P<date>(\\d\\d){0,2})')
```

Comment:
Do you need the double `\`? You shouldn't need them right?

Reply:
the double \ is needed inside a python string, unlike in the yaml definition in the other repo. python is a little weird in that it ignores unrecognised escape sequences (like `\d`), but that's a fudge and might at some point in the future become a valid escape sequence. it's safer to double back-slash or to use a raw string

Reply:
https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-stringescapeseq